kernel/src/analysis/analyze.rs

```diff
@@ -47,70 +47,72 @@ pub fn analyze(
 ) -> Vec<RuleResult> {
     let lines_to_ignore = get_lines_to_ignore(code, language);
 
-    match get_tree(code, language) {
-        Some(tree) => rules
-            .into_iter()
-            .map(|rule| {
-                let invalid_query_result = RuleResult {
-                    rule_name: rule.name.clone(),
-                    filename: filename.to_string(),
-                    violations: vec![],
-                    errors: vec![ERROR_INVALID_QUERY.to_string()],
-                    execution_error: None,
-                    execution_time_ms: 0,
-                    output: None,
-                };
+    get_tree(code, language).map_or_else(
```

Comment:
This one is possibly the most subjective one although it's also based on some clippy lints like [this](https://rust-lang.github.io/rust-clippy/master/index.html#/option_if_let_else) and [this one](https://rust-lang.github.io/rust-clippy/master/index.html#/option_if_let_else).

Reply:
I agree with you, this one does not really make the code any better. I would tend to not include this change as its quite hard to read. An alternative would be to change the mapping code and divide it into functions to make it more digestible.